pkg/controller/kappnav/kappnav_controller.go

```diff
@@ -450,21 +572,59 @@ func (r *ReconcileKappnav) Reconcile(request reconcile.Request) (reconcile.Resul
 			Namespace: instance.GetNamespace(),
 		},
 	}
-	err = r.CreateOrUpdate(controllerDeployment, instance, func() error {
+	if (logger.IsEnabled(kappnavutils.LogTypeInfo)) {
+		logger.Log(kappnavutils.CallerName(), kappnavutils.LogTypeInfo, "Create or update controller deployment" + otherLogData, logName)
+	} 	
+	err = r.CreateOrUpdate(logger, controllerDeployment, instance, func() error {
 		pts := &controllerDeployment.Spec.Template
 		kappnavutils.CustomizeDeployment(controllerDeployment, instance)
 		kappnavutils.CustomizePodSpec(pts, &controllerDeployment.ObjectMeta,
 			kappnavutils.CreateControllerDeploymentContainers(pts.Spec.Containers, instance), nil, instance)
 		return nil
 	})
 	if err != nil {
-		reqLogger.Error(err, "Failed to reconcile the Controller Deployment")
-		return r.ManageError(err, kappnavv1.StatusConditionTypeReconciled, instance)
-	}
+		if (logger.IsEnabled(kappnavutils.LogTypeError)) {
+			logger.Log(kappnavutils.CallerName(), kappnavutils.LogTypeError, fmt.Sprintf("Failed to reconcile the Controller Deployment" + otherLogData + ", Error: %s", err), logName)
+		}
+		return r.ManageError(logger, err, kappnavv1.StatusConditionTypeReconciled, instance)
+	}	
 
 	// If an extension exists call its reconcile function, otherwise return success.
 	if extension != nil {
-		return extension.ReconcileAdditionalResources(request, &r.ReconcilerBase, instance)
+		return extension.ReconcileAdditionalResources(logger, request, &r.ReconcilerBase, instance)
```

Comment:
Check whether you also updated the interface. I don't see that in the PR. The interface's function requires the Logger parameter, otherwise I wouldn't expect this to compile.
```
type KappnavExtension interface {
	ReconcileAdditionalResources(request reconcile.Request, r *ReconcilerBase, instance *kappnavv1.Kappnav) (reconcile.Result, error)
}
```

Reply:
done